src/data_structures/suffix_array.rs

```diff
@@ -6,6 +6,17 @@
 //! Suffix arrays and related algorithms.
 //! The implementation is based on the lecture notes
 //! "Algorithmen auf Sequenzen", Kopczynski, Marschall, Martin and Rahmann, 2008 - 2015.
+//! # Example
```

Comment:
Maybe add some suffix array link and/or the original citation for the suffix array to the docs?